### PR TITLE
Update libjxl dependencies, fix static compilation

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Description of changes

Updates packages and adds special options to make them work with `pkgsStatic`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libwebp/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwebp/default.nix:21]:
```suggestion
```
would be great if we could drop this

File: pkgs/development/libraries/libwebp/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwebp/default.nix:33]:
```suggestion
  postPatch = "patchShebangs .";
```
can we more specific about the files which need patching?

File: pkgs/development/libraries/openexr/3.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/openexr/3.nix:24]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/libraries/libwebp/default.nix
Comment (dali99) [reply pkgs/development/libraries/libwebp/default.nix]: It seems it could be completely removed

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by alyssais** [comment]
> The most objectionable thing here is probably removing `gperftools` in libjxl when statically compiling but it seems to work tm

I was able to get `pkgsStatic.gperftools` to build by removing the `libunwind` dependency, as is already done for ARM platforms.  That should allow `libjxl` to have its `gperftools` dependency even when building statically.

**Comment by dali99** [comment]
While trying to apply what was said in https://github.com/NixOS/nixpkgs/pull/179102#issuecomment-1171763597 I got some weird compilation errors I didn't see before (even without those changes). I dont have time to check out why right now, so please hold off on merging for a few more days. Converting to draft in the mean-time

**Comment by alyssais** [comment]
> I got some weird compilation errors I didn't see before (even without those changes). I dont have time to check out why right now

Can you share what they are?  It's likely stuff that's just broken on staging in the meantime and is already a known issue — it's been quite volatile recently.

**Comment by alyssais** [comment]
The fix for `pkgsStatic.libwebp` (with TIFF support) is https://github.com/NixOS/nixpkgs/pull/180018.

**Comment by alyssais** [comment]
@dali99 want to try rebasing on the latest staging and seeing if pkgsStatic.libwebp can build with libtiff support?

**Comment by alyssais** [comment]
@OfBorg build pkgsStatic.libjxl

**Comment by dali99** [comment]
@ofborg build pkgsStatic.libjxl libjxl

**Comment by alyssais** [comment]
What do you think of this patch, instead of disabling libhwy's tests?

```diff
From f9f58d5540b91179d246f36dd674b0afd92bbdd9 Mon Sep 17 00:00:00 2001
From: Alyssa Ross <hi@alyssa.is>
Date: Sat, 9 Jul 2022 06:54:33 +0000
Subject: [PATCH nixpkgs] libhwy: fix tests when cross-compiling

It seems to need some help to find gtest when cross compiling.

(Only applies when cross-compiling to compatible architectures where
the tests are enabled, like pkgsStatic.)
---
 pkgs/development/libraries/libhwy/default.nix | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/pkgs/development/libraries/libhwy/default.nix b/pkgs/development/libraries/libhwy/default.nix
index 7656e063822..68c7b5affa8 100644
--- a/pkgs/development/libraries/libhwy/default.nix
+++ b/pkgs/development/libraries/libhwy/default.nix
@@ -1,5 +1,13 @@
 { lib, stdenv, cmake, ninja, gtest, fetchpatch, fetchFromGitHub }:

+let
+  libExt =
+    if stdenv.hostPlatform.isStatic then
+      ".a"
+    else
+      stdenv.hostPlatform.extensions.sharedLibrary;
+in
+
 stdenv.mkDerivation rec {
   pname = "libhwy";
   version = "0.15.0";
@@ -23,8 +31,6 @@ stdenv.mkDerivation rec {

   nativeBuildInputs = [ cmake ninja ];

-  checkInputs = [ gtest ];
-
   # Required for case-insensitive filesystems ("BUILD" exists)
   dontUseCmakeBuildDir = true;

@@ -32,7 +38,12 @@ stdenv.mkDerivation rec {
     "-GNinja"
     "-DCMAKE_INSTALL_LIBDIR=lib"
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
-  ] ++ lib.optional doCheck "-DHWY_SYSTEM_GTEST:BOOL=ON";
+  ] ++ lib.optionals doCheck [
+    "-DHWY_SYSTEM_GTEST:BOOL=ON"
+    "-DGTEST_INCLUDE_DIR=${lib.getDev gtest}/include"
+    "-DGTEST_LIBRARY=${lib.getLib gtest}/lib/libgtest${libExt}"
+    "-DGTEST_MAIN_LIBRARY=${lib.getLib gtest}/lib/libgtest_main${libExt}"
+  ];

   # hydra's darwin machines run into https://github.com/libjxl/libjxl/issues/408
   doCheck = !stdenv.hostPlatform.isDarwin;
--
2.35.1
```

File: pkgs/development/libraries/libhwy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libhwy/default.nix:8]: We should move this next to sharedLibrary in lib.

File: pkgs/development/libraries/gperftools/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/gperftools/default.nix:32]: FYI

```suggestion
  buildInputs = lib.optional (stdenv.isLinux && !(stdenv.isAarch || stdenv.hostPlatform.isStatic)) libunwind;
```

after #183283 would be merged

**Comment by alyssais** [comment]
[nitpick] The commit summary "lib/systems: Add common static library extensions to extensions, and common attribute for static and shared libraries" is really long.  A more reasonable one might be "lib/systems: add staticLibrary and library", with more explanation in the body.  As it is, GitHub will truncate it, and so will git log unless somebody has a really wide terminal.

But I'm not going to hold up the PR any more for that, just something to be aware of in future.

File: pkgs/development/libraries/libhwy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libhwy/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/libhwy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libhwy/default.nix:33]:
```suggestion
  cmakeFlags = let
    libExt = stdenv.hostPlatform.extensions.library;
  in
  [
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM
There are some ofborg failures. Please take a look at them.

**Comment by alyssais** [comment]
@OfBorg build pkgsStatic.libjxl

**Comment by roconnor-blockstream** [comment]
@dali99 Is there some way to patch `pprof` shebangs so that it refers to a fixed perl derivation?  I'd like to run `pprof` in a pure environment.

**Comment by dali99** [comment]
I don't see what that has to do with this PR, I'm also not really an expert in perl and haven't used pprof. I'm also not sure which pprof shebangs you're asking about, so I don't think this is the right place to talk about this.

Sorry I can't be of anymore help, maybe you can try asking in [#nix:nixos.org](https://matrix.to/#/#nix:nixos.org) or on the [discourse](https://discourse.nixos.org/c/learn/9)?

**Comment by roconnor-blockstream** [comment]
Sorry you are right.  This was the last PR to bump the version of gperftools.  I didn't realize you were not the maintainer.

**Pull request merged**
